packages/slice-machine/src/modules/selectedSlice/reducer.ts

```diff
@@ -0,0 +1,273 @@
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+      };
+    }
+    case getType(generateSliceScreenshotCreator): {
+      if (!prevState) return prevState;
+
+      const newModel = {
+        ...prevState.Model,
+        screenshotUrls: action.payload.screenshots,
+      };
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+      };
+    }
+    case getType(generateSliceCustomScreenshotCreator): {
+      if (!prevState) return prevState;
+      const { variationId, screenshot } = action.payload;
+      const { Model: prevModel } = prevState;
+
+      const screenshots = prevModel.variations.reduce((acc, variation) => {
+        if (variation.id === variationId) {
+          return {
+            ...acc,
+            [variationId]: screenshot,
+          };
+        }
+        if (prevModel.screenshotUrls?.[variation.id]) {
+          return {
+            ...acc,
+            [variation.id]: prevModel.screenshotUrls?.[variation.id],
+          };
+        }
+        return acc;
+      }, {});
+
+      const newModel = { ...prevModel, screenshotUrls: screenshots };
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+      };
+    }
+    case getType(saveSliceCreator): {
+      if (!prevState) return prevState;
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(action.payload.state),
+      };
+    }
+    case getType(pushSliceCreator): {
+      if (!prevState) return prevState;
+      const newModel = {
+        ...prevState.Model,
+        __status: LibStatus.Synced,
+        initialScreenshotUrls: prevState.Model.screenshotUrls,
+        remoteVariations: prevState.Model.variations,
+      };
+
+      return { ...prevState, Model: updateTouchedAndStatus(newModel) };
+    }
+    case getType(copyVariationSliceCreator): {
+      if (!prevState) return prevState;
+      const { key, name, copied } = action.payload;
+      const newVariation = Variation.copyValue(copied, key, name);
+      const newModel = {
+        ...prevState.Model,
+        variations: prevState.Model.variations.concat([newVariation]),
+      };
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+        variation: newVariation,
+      };
+    }
+    default:
+      return prevState;
+  }
+};
+
+const updateTouchedAndStatus = (model: SliceState) => {
+  const isTouched =
+    !equal(model.initialVariations, model.variations) ||
+    !equal(model.initialMockConfig, model.mockConfig);
+  let status = model.__status;
+
+  const isScreenshotModified = !equal(
+    model.screenshotUrls,
+    model.initialScreenshotUrls
+  );
+
+  // True if the remote and local slice variations don't match
+  const isModelModified = !compareVariations(
+    model.remoteVariations,
+    model.initialVariations
+  );
+
+  if (isModelModified || isScreenshotModified) {
+    status = LibStatus.Modified;
```

Comment:
If it's a new Slice and there is nothing on remote, wouldn't `isModelModfied` be true and the status displayed be `Modified` and not `New` ?

Reply:
uff good one to spot! I fixed it here but while manually testing realise we have another issue related to it:

We update the __status in the `selectedSlice` redux state but not in the `library` state, which means when you go back to the previous page (where the status is shown in the card), it does not update unless you refresh the page :(

Reply:
not too sure what the best way of fixing this is tho...

Reply:
guess we could listen to all redux actions of type sliceCreated, sliceSaved and slicePushed inside the library redux, and when those are triggered we can update the `__status` of the corresponding slice? What do you think

Reply:
Discussed during daily, we'll create a new ticket and tackle it right away :)